### PR TITLE
JSON-RPC API doesn't support HD Wallet

### PR DESCRIPTION
**Pull request by r8d8**
Added calls fro import/export, signTransaction
Related to #195

### CODE REVIEW

**Comment by r8d8** [comment]
Not sure how se should deal with `hide/unhide` rpc calls

File: docs/api.adoc [hunk 1]
```diff
@@ -260,6 +260,36 @@ Import a new account from an external keyfile.
 <-- {"jsonrpc": "2.0", "result": "0x0047201aed0b69875b24b614dda0270bcd9f11cc", "id": 1}
 ----
 
+=== emerald_importHDAccount
```
Comment (dulanov) [inline docs/api.adoc]: Why not to add additional meta attribute like `hardware_wallet` to `emerald_importAccount`? Why you need definitely a new JSON-RPC method instead?

Comment (r8d8) [inline docs/api.adoc]: `Keyfile`  for hd wallet differs a lot from normal one.

Comment (r8d8) [inline docs/api.adoc]: But, yeah, we can just inject handling for HD_Keyfile inside original callback
Thanks

File: docs/api.adoc [hunk 2]
```diff
@@ -44,7 +44,7 @@ Returns the client current version.
 
 Return the list of all not hidden (by default) accounts from the keystore.
 
-*Parameters*:
+*Parameters*:b
```
Comment (dulanov) [inline docs/api.adoc]: Remove here `b`

File: docs/api.adoc [hunk 3]
```diff
@@ -503,6 +546,7 @@ If `function` and `arguments` are provided, they will be encoded according smart
     * `additional` (Object, optional)
     ** `chain` (String, optional) - chain name, by default `mainnet`, other possible variant `testnet`
     ** `chain_id` (Number, optional) - chain id number, by default for `mainnet` it equals `61`
+    ** `hd_path` (String, optional) - derivation path for signing with HD wallet, by default empty
```
Comment (dulanov) [inline docs/api.adoc]: Did you confuse method? `emerald_listAccounts` instead?

File: docs/api.adoc [hunk 4]
```diff
@@ -56,6 +56,7 @@ Return the list of all not hidden (by default) accounts from the keystore.
     * `accounts` (Array)
     ** `account` (Object) - an account
     *** `address` (String) - hex-encoded 20 bytes public address
+    *** `is_hd` (Boolean) - flag to distinguish normal accounts from HD wallet accounts
```
Comment (dulanov) [inline docs/api.adoc]: A little bit cryptic, not good for API, probably better to use `hardware_wallet`?

File: docs/api.adoc [hunk 5]
```diff
@@ -56,6 +56,7 @@ Return the list of all not hidden (by default) accounts from the keystore.
     * `accounts` (Array)
     ** `account` (Object) - an account
     *** `address` (String) - hex-encoded 20 bytes public address
+    *** `hardware_wallet` (Boolean) - flag to distinguish normal accounts from HD wallet accounts
```
Comment (dulanov) [inline docs/api.adoc]: Sorry, but may be `hd_wallet`, I really don't know how better to name? ;)

File: docs/api.adoc [hunk 6]
```diff
@@ -260,6 +270,22 @@ Import a new account from an external keyfile.
 <-- {"jsonrpc": "2.0", "result": "0x0047201aed0b69875b24b614dda0270bcd9f11cc", "id": 1}
 ----
 
+* HD wallet account:
+----
+--> {"jsonrpc": "2.0",
+     "method": "emerald_importAccount",
+     "params":
+       [{"version": 3,
+         "id": "f7ab2bfa-e336-4f45-a31f-beb3dd0689f3",
```
Comment (dulanov) [inline docs/api.adoc]: I don't see here declared before `keyfile` field, is it ok?

Comment (dulanov) [inline docs/api.adoc]: Ah, ok, I see now, forget.

**Pull request merged**